### ISSUE TITLE
IJ-digraph

### ISSUE THREAD

**Issue by mikkelee** [georgd/EB-Garamond#133]
It would be nice to have IJ-digraphs for Dutch and other texts which use it.

These are some quick and ugly versions I made, I can supply them in vector format if needed as a base (.eps?)
![ij](https://user-images.githubusercontent.com/644671/196930458-d53d9320-774a-4bfc-85c3-d41d94c53373.png)

**Comment by georgd** [comment]
Agreed. The italics ligature needs to be added. However, I doubt that the roman ligature would look good. There are no vertical stems as narrowly set next to each other throughout the font. Such glyph would stand out as a black blob in every text.

**Comment by mikkelee** [comment]
Yeah, it was just a quick sketch.

This version only has digraph for italics, but not sure what Garamond variant is used:
https://commons.wikimedia.org/wiki/File:Comparison_of_normal_and_italic_of_ij_and_ÿ.svg

**Comment by Alexander-Wilms** [comment]
There's an italic ij ligature in the EB specimen:

_Cursiff Parangon de Granjon_, line 4:

![image](https://user-images.githubusercontent.com/3226457/225812145-3dd4bb0a-bcd3-4104-9203-55a5b0c59126.png)

Do you have a higher resolution specimen available? I only found https://image.linotype.com/files/pdf/specimen.pdf and https://archive.org/details/historytechnique0000nesb/page/100/mode/2up

**Comment by mikkelee** [comment]
The scanned version at the Universitätsbibliothek Frankfurt am Main is quite high resolution:
[link](http://sammlungen.ub.uni-frankfurt.de/urn/urn:nbn:de:hebis:30:2-246255)

Note that the available PDF is quite small, but if you click through to full screen in your browser, you can zoom.
![3](https://user-images.githubusercontent.com/644671/225824429-d10eda32-315a-4b07-8821-aaf186b3aa0c.jpeg)

**Comment by georgd** [comment]
Thank you, both! Meanwhile, I have another very good source these fonts. Henrik D. L. Vervliet (2010) “French renaissance printing types: a conspectus” is a comprehensive collection on the topic. The ij ligature is definitely well attested in Granjon’s italics of all sizes :)

I just don’t see a good example for this ligature in a roman font.

**Comment by mikkelee** [comment]
Likely Granjon never created a roman minuscule digraph/ligature — possibly it's not really a thing? In modern Dutch typography, it seems to mostly show up in capitals and italics, though some typefaces might use kerning (as in my bad example)...

Disclaimer: I'm not Dutch, but I transcribe a lot of pre-1800 Danish cursive (so-called "Gothic cursive"), where it was sometimes used. In those sources it looks pretty much identical to _ÿ_ (and indeed _y_ was usually written with diaresis, so distinguishing _ÿ/y_ and _ij_ is mostly contextual, based on what you know the spelling "should" be). So for my sake, there's not really a need for a roman ij-digraph, as I use italics when I quote sources that use them ;)

https://en.wikipedia.org/wiki/IJ_(digraph)

**Comment by georgd** [comment]
Oh, now I now why I was so confused when you said there was no i_j ligature in the font because I was sure I did it. But then again couldn’t find it. Being away from that for almost ten years really makes one forget things. There’s of course a code point in Unicode (U+0133) and it is populated in all four fonts (EBG12 Re and It and EBG08 Re and It). They’re very tightly set in the regular fonts—just as tightly as I before said I wouldn’t do it (oh my)—and the 08-It one even is really ligated whereas the one in 12-It isn’t. What’s missing altogether is a lookup for ligature replacement besides using the dedicated code point.

So the todos are actually

- [ ] draw a ligated variant in 12-It for U+0133
- [ ] review U+0133 in 08-It
- [ ] copy references to U+0133 to an unencoded slot `i_j` in all fonts
- [ ] add a lookup for `sub i j by i_j` (localized default liga for nld and/or discretionary for all latin script?)
- [ ] add handling for accented íj́

**Comment by timgrei** [comment]
@georgd Does the specimen reflect the current state of the project (apart from initials)?

**Comment by mikkelee** [comment]
Thank you Georg, that looks like an excellent path forward. I will help as much as I can!

Tim: If you mean the specimen.pdf from this project it does not. The nightly updates just rebuild the font files, everything else stays as is until another manual release

**Comment by timgrei** [comment]
Yes, I know. My wording was not clear: Does it reflect the state of the project, before we started working on it again.

**Comment by mikkelee** [comment]
Sorry, I understand :) I think this is the latest version online from before the new interest. Both the file modification date and the front page are dated Apr 8, 2014
https://github.com/georgd/EB-Garamond/blob/82c892d3b1ecf6fe52554c5133af9127974b9be8/specimen/Specimen.pdf

**Comment by georgd** [comment]
Ok. The specimen is indeed on the state of 2014. The current nightly adds lots of Initials (thanks @timgrei ) and reflects the beginning of remedying some overzealousness of mine: 

- German orthotypography commanded a linguistically determined use of ligatures which is no longer reflected in dictionaries and is sometimes regarded as antiquated. So, the sophistically coded feature shall be easily switchable.
- I generalised the renaissance use of the glyphs V/u for the Latin locale. This is indeed to be considered as buggy. It should be optional only.

**Comment by moyogo** [comment]
@mikkelee

There are words in Dutch texts that have ij as i and j  instead of the digraph ij.
Some of these are compound words or words with affixes and should be spelled with i-j instead of ij when following the 2006 spelling: groeijaar, minijurk, strooijonker, etc.
Some are borrowed words: bijectie, bijou, dijonmosterd, millijoule, etc. Some of these also have a specific official spelling: the common hijab is officially hidjab.
Some are foreign place or topographic names and their derivatives: Beijing, Fiji (Fijiër, ...), Rijeka, etc. and many given names: Marija, Vijay, etc. or family names: Dostojevskij, Seijas, etc.
There also odd ones like hoeladije or tararaboemdijee.

In a few cases it can be both Dutch ij or foreign i+j depending on the context, like Dutch given name Arijan or Serbian given name Arijan, or Dutch toponym Meije and French toponym La Meije.

There’s no problem having a proper ij ligature for those in context where ligatures are common, like in an italic that has many ligatures or in handwritten styles. In roman styles it’s more problematic, if present it should mostly likely be in dlig or  ssXX, or at least easy to disable.

@georgd
For accented íj́, the current spelling rule for marking stress was settled in the 1996 spelling. Before that it was common to put only one acute on digraphs composed of different letters. Most texts from before 1996 and a lot of more recent text still follow pre-1996 rules with ["níet"](https://www.google.com/search?q=%22n%C3%ADet%22+-%22n%C3%AD%C3%A9t%22+%28site%3A.nl+OR+site%3Abe%29&oq=%22n%C3%ADet%22+-%22n%C3%AD%C3%A9t%22+%28site%3A.nl+OR+site%3Abe%29) instead of ["níét"](https://www.google.com/search?q=%22n%C3%AD%C3%A9t%22+-%22n%C3%ADet%22+%28site%3A.nl+OR+site%3Abe%29&oq=%22n%C3%AD%C3%A9t%22+-%22n%C3%ADet%22+%28site%3A.nl+OR+site%3Abe%29). The text of the [spelling rules](https://woordenlijst.org/leidraad/5/1) is very clear: "Because of technical limitations the acute on j of the lange ij is usually omitted" and provides an example "blíjven kijken!". There are also foreign names in Dutch texts with íj where the acute on j is omitted on purpose like Níjar or Szíjj.
The only way to know if the j should have an acute after í is if it is used with the combining acute character. iOS/macOS and Android Dutch keyboard layouts let users input j́ just like í since 2019.

**Comment by georgd** [comment]
@moyogo thanks a lot for your valuable comment. 

I think there’s nothing special to do for a Dutch localization

- The italics may use a ligated `i_j` regardless of the localization by means of the `dlig` feature.
- The encoded ij in `U+0133` has the ligated form by default (in italics), perhaps to be switched off by CVxx (is this necessary?)
- Uppercase IJ in italics and upper and lowercase IJ/ij in the regular fonts stay unchanged.
- `íj́` should be handled like accented glyphs in general without special guesswork.

Would you agree?

**Comment by moyogo** [comment]
> Would you agree?

Yes, that sounds good to me.

Some users may appreciate a ligated /ij in Roman and /i_j or a special /IJ and /I_J  as a baseline J with a shortened I in the top left, for signage or display for example, as a ssXX stylistic set or character variant cvXX feature. But for text this is rather unusual and breaks the examples mentionned above.

**Comment by georgd** [comment]
> > Would you agree?
> 
> Yes, that sounds good to me.
> 
> Some users may appreciate a ligated /ij in Roman and /i_j or a special /IJ and /I_J as a baseline J with a shortened I in the top left, for signage or display for example, as a ssXX stylistic set or character variant cvXX feature. But for text this is rather unusual and breaks the examples mentionned above.

Thanks! I don't see how the display I_J you’re referring to could match this font’s design. I think this works fine in fonts that show a wider bottom arc in upper case J. Next to the narrow Garamond J the U shaped I_J feels much out of place, in my opinion.

**Comment by KrasnayaPloshchad** [comment]
> There are words in Dutch texts that have ij as i and j instead of the digraph ij.
> Some of these are compound words or words with affixes and should be spelled with i-j instead of ij when following the 2006 spelling: groeijaar, minijurk, strooijonker, etc.

Such thing can be made by inserting ZWNJ between i and j to prevent ligature.

**Comment by moyogo** [comment]
> > There are words in Dutch texts that have ij as i and j instead of the digraph ij.
> > Some of these are compound words or words with affixes and should be spelled with i-j instead of ij when following the 2006 spelling: groeijaar, minijurk, strooijonker, etc.
> 
> Such thing can be made by inserting ZWNJ between i and j to prevent ligature.

Nobody (in the pool of normal users) is going to insert ZWNJ in Dutch text. The consensus is that ij is a digraph that sometimes behaves like a single letter, at least in the official spelling rules. The ligature should not be the default but should be easily accessible for when it makes sense.

**Comment by KrasnayaPloshchad** [comment]
So you need to make an investigation on Dutch dictionaries. So you can make analysis to acknowledge which words shouldn’t have ij ligature and set rules for them.

**Comment by moyogo** [comment]
> So you need to make an investigation on Dutch dictionaries.

What do you mean?

**Comment by moyogo** [comment]
> So you need to make an investigation on Dutch dictionaries.

@KrasnayaPloshchad What do you mean?
